src/MySqlConnector/Serialization/ColumnDefinitionPayload.cs

```diff
@@ -35,17 +51,90 @@ public static ColumnDefinitionPayload Create(PayloadData payload)
 			}
 
 			if (reader.BytesRemaining != 0)
+			{
 				throw new FormatException("Extra bytes at end of payload.");
-			return new ColumnDefinitionPayload(name, characterSet, columnLength, columnType, columnFlags);
+			}
+
+			return new ColumnDefinitionPayload
+			{
+				Name = name,
+				CharacterSet = characterSet,
+				ColumnLength = columnLength,
+				ColumnType = columnType,
+				ColumnFlags = columnFlags,
+				SchemaName = schema,
+				CatelogName = catalog,
+				Table = table,
+				PhysicalTable = physicalTable,
+				PhysicalName = physicalName,
+				Decimals = decimals
+			};
+
 		}
 
-		private ColumnDefinitionPayload(string name, CharacterSet characterSet, int columnLength, ColumnType columnType, ColumnFlags columnFlags)
+		public Type GetDataType(
+			bool treadTinyAsBoolean = false,
```

Comment:
Typo: `treatTinyAsBoolean`

Reply:
Seems to be copied/duplicated from `ResultSet.GetFieldType`; that method should probably be refactored to delegate to this one.

Reply:
Fixed by use `ResultSet.GetFieldType`